src/include/duckdb/execution/operator/join/physical_hash_join.hpp

```diff
@@ -52,8 +50,9 @@ class PhysicalHashJoin : public PhysicalComparisonJoin {
 
 	//! Duplicate eliminated types; only used for delim_joins (i.e. correlated subqueries)
 	vector<LogicalType> delim_types;
-	//! Used in perfect hash join
-	PerfectHashJoinStats perfect_join_statistics;
+
+	//! Join Keys statistics (optional)
+	vector<unique_ptr<BaseStatistics>> join_stats;
```

Comment:
Maybe I'm missing something but it seems like this is never appended to, which would result in never triggering the fallback?